crates/api/src/relay_data/api.rs

```diff
@@ -131,15 +141,48 @@ impl<DB: DatabaseService + 'static> DataApi<DB> {
         }
     }
 
-    pub async fn constraints() {
-        unimplemented!()
+    /// Implements this API: <https://chainbound.github.io/bolt-docs/api/relay#constraints>
+    pub async fn constraints(
+        Extension(data_api): Extension<Arc<DataApi<A, DB>>>,
+        Query(params): Query<usize>,
+    ) -> Result<impl IntoResponse, DataApiError> {
+        // TODO: Return err if the slot is outside expiry bounds i.e. before an epoch
+        // For this we need a beacon state tracker
+        let slot = params as u64;
```

Comment:
It makes sense to return an error in such case. Anyway, I'd consider to make the `slot` parameter optional (and reflect it in the docs), and when it is not sent we should consider the current slot (from the beacon state) the default one. It is most likely the most requested slot.

Reply:
Good point